### PR TITLE
BVAL-682 Misc. clarifications around container element validation

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://hibernate.atlassian.net/projects/BVAL/issues/BVAL-682

### CODE REVIEW

File: sources/exception.asciidoc [hunk 1]
```diff
@@ -157,29 +157,15 @@ Reasons for raising this exception include:
 * The extracted type is not marked with `@ExtractedValue`
 * The `@ExtractedValue` annotation is given more than once for one value extractor type
 
-[role="api"]
```
Comment (gunnarmorling) [inline sources/exception.asciidoc]: I've removed the includes to be consistent with what we're doing for the other exceptions.

@gsmet Should we sprinkle some tck-testable markers over the examples given for exception cases?

Comment (gsmet) [inline sources/exception.asciidoc]: I don't think it's a good idea. They are only examples. The tck-testable assertions should be present in the according section.

File: sources/validation-api.asciidoc [hunk 2]
```diff
@@ -529,7 +529,7 @@ It is possible to narrow a node instance to its precise type and extract node sp
 [tck-testable]
 --
 `BeanNode`, `PropertyNode` and `ContainerElementNode` host `getContainerClass()` and `getTypeArgumentIndex()`.
-If the node represents an element that is contained in a generic container such as `Optional`, `List` or `Map`,
+If the node represents an element that is contained in a container such as `Optional`, `List` or `Map`,
 the former returns the declared type of the container and the latter returns the index of the affected type argument.
```
Comment (gsmet) [inline sources/validation-api.asciidoc]: Why did you remove the `generic` word? We are talking about the type argument so it only makes sense for generic containers.

File: sources/value-extractor-definition.asciidoc [hunk 3]
```diff
@@ -106,8 +110,8 @@ Compatible implementations provide value extractors for the following types out
...
-* [tck-testable]#`java.util.Optional`; `value()` must be invoked (passing the contained value or `null` if none is present), passing `null` as node name#
-* [tck-testable]#`java.util.OptionalInt`, `java.util.OptionalLong` and `java.util.OptionalDouble`; the extracted value types must be `java.lang.Integer`, `java.lang.Long` and `java.lang.Double`, respectively. `value()` must be invoked (passing the contained value or `null` if none is present), passing `null` as node name. The extractors must be marked with `@UnwrapByDefault`.#
+* [tck-testable]#`java.util.Optional`; `value()` must be invoked, passing `null` as node name and passing the contained object as value or `null` if none is present)#
+* [tck-testable]#`java.util.OptionalInt`, `java.util.OptionalLong` and `java.util.OptionalDouble`; the extracted value types must be `java.lang.Integer`, `java.lang.Long` and `java.lang.Double`, respectively. `value()` must be invoked, passing `null` as node name and passing the contained number as value or `null` if none is present). The extractors must be marked with `@UnwrapByDefault`.#
```
Comment (gsmet) [inline sources/value-extractor-definition.asciidoc]: `present)` I can't find the starting parenthesis so I suppose we should remove it?

File: sources/value-extractor-definition.asciidoc [hunk 4]
```diff
@@ -106,8 +110,8 @@ Compatible implementations provide value extractors for the following types out
 * [tck-testable]#`java.util.Map`; both map keys and map values are to be supported;
 `keyedValue()` must be invoked by the map key extractor for each contained key, passing the string literal `<map key>` as node name;
 `keyedValue()` must be invoked by the map value extractor for each contained value, passing the string literal `<map value>` as node name#
-* [tck-testable]#`java.util.Optional`; `value()` must be invoked (passing the contained value or `null` if none is present), passing `null` as node name#
-* [tck-testable]#`java.util.OptionalInt`, `java.util.OptionalLong` and `java.util.OptionalDouble`; the extracted value types must be `java.lang.Integer`, `java.lang.Long` and `java.lang.Double`, respectively. `value()` must be invoked (passing the contained value or `null` if none is present), passing `null` as node name. The extractors must be marked with `@UnwrapByDefault`.#
+* [tck-testable]#`java.util.Optional`; `value()` must be invoked, passing `null` as node name and passing the contained object as value or `null` if none is present)#
```
Comment (gsmet) [inline sources/value-extractor-definition.asciidoc]: present) I can't find the starting parenthesis so I suppose we should remove it?

File: sources/exception.asciidoc (hunk 1 above)
Comment (gunnarmorling) [inline sources/exception.asciidoc]: Ok. Note these are not "examples" in the usual sense, but some of the specific cases where this exception is to be thrown. But I think it's covered in the specific sections, too.

File: sources/validation-api.asciidoc (hunk 2 above)
Comment (gunnarmorling) [inline sources/validation-api.asciidoc]: I removed it to cover the case where `getContainerClass()` returns the container class for a non-generic container. `getTypeArgumentIndex()` returns null in that case as per its documentation. I'll add "the index of the affected type argument if the container is of a generic type".

File: sources/value-extractor-definition.asciidoc (hunk 3 above)
Comment (gunnarmorling) [inline sources/value-extractor-definition.asciidoc]: Good spot.

**Comment by gunnarmorling** [comment]
@gsmet thanks a lot for your remarks - reworked accordingly and force-pushed the commit.
